.github/workflows/update-ami.py

```diff
@@ -1,51 +1,57 @@
-#!/usr/bin/python3
+#!/usr/bin/env python3
 import json
 import os 
 import boto3
 import uuid
 import re
-import packer
+import subprocess
 
-uid = str(uuid.uuid4())
-file = '.github/workflows/amichange.json'
-packerfile = ".github/packer/ubuntu-focal-x86_64-public-ami.json"
+# Globals
+amifile = '.github/workflows/amichange.json'
+packerfile = ".github/packer/ubuntu-jammy-x86_64-public-ami.json"
 
-update_marketplace = True
+# Environment Globals
 product_id = os.getenv('PRODUCT_ID')
 role_arn = os.getenv('ROLE_ARN')
 vtag = os.getenv('TAG')
 tag = vtag.replace('v', '')
-variables = [product_id,role_arn,tag]
+skip_create_ami = os.getenv('SKIP_CREATE_AMI', True)
 
-for var in variables:
-  if var is None:
-    print("A Variable is not set correctly or this is not the right repo.  Only validating packer.")
-    update_marketplace = False
+def packer_build(packerfile):
+  print("Running the packer build")
+  subprocess.run('/usr/local/bin/packer build ' + packerfile, shell=True)
 
-if 'rc' in tag or 'fuji' in tag:
-  print("This is a release candidate.  Only validating packer.")
-  update_marketplace = False
 
-def packer_build(packerfile):
-  p = packer.Packer(packerfile)
-  output = p.build(parallel=False, debug=False, force=False)
-  found = re.findall('ami-[a-z0-9]*', str(output))
-  return found[-1]
+def packer_build_update(packerfile):
+  print("Creating packer AMI image for Marketplace")
+  output = subprocess.run('/usr/local/bin/packer build ' + packerfile, shell=True, stdout=subprocess.PIPE)
+  found = re.findall('ami-[a-z0-9]*', str(output.stdout))
+
+  if found:
+    amiid = found[-1]
+    return amiid
+  else:
+    raise RuntimeError(f"No AMI ID found in packer output: {output.stdout}")
 
-def parse_amichange(object):
-  with open(object, 'r') as file:
+def parse_amichange(amifile, amiid, role_arn, tag):
+  # Create json blob to submit with the catalog update
+  print("Updating the json artifact with recent amiid and tag information")
+  with open(amifile, 'r') as file:
     data = json.load(file)
 
   data['DeliveryOptions'][0]['Details']['AmiDeliveryOptionDetails']['AmiSource']['AmiId']=amiid
   data['DeliveryOptions'][0]['Details']['AmiDeliveryOptionDetails']['AmiSource']['AccessRoleArn']=role_arn
   data['Version']['VersionTitle']=tag
   return json.dumps(data)
 
-amiid=packer_build(packerfile)
-
-if update_marketplace:
-
+def update_ami(amifile, amiid):
+  # Update the catalog with the last amiimage
+  print('Updating the marketplace image')
   client = boto3.client('marketplace-catalog',region_name='us-east-1')
+  uid = str(uuid.uuid4())
+  global tag
+  global product_id
+  global role_arn
```

Comment:
Sorry for just - not knowing python... but why do we make these `global`? They are already in scope from defining them outside of the function definitions... no?

Reply:
Yea, so I had an issue where it was not pulling in the variables to the function unless supplied.  I could have added them as options to the function but this was the recommend python way.

Reply:
I'll run another test now that I worked out the other bugs and see if it's still necessary

Reply:
Yea, just tested and we need that.  OTherwise it can not find the role_arn or product_id and fails.